src/Collection.php

```diff
@@ -376,12 +376,12 @@ public function frequency(): CollectionInterface
      *
      * @param mixed ...$parameters
      */
-    public static function fromCallable(callable $callable, ...$parameters): self
+    public static function fromCallable(callable $callable, ...$parameters): CollectionInterface
```

Comment:
will there be any issues with this change? I'm thinking the only potential issues might be due to the interface and the class not being fully in sync when it comes to annotations, but functionality should be impacted

Reply:
I always use `self` in conjunction of a static constructor. To me, it makes more sense.

Could you please give me more details on this change? What's the reason?

Reply:
I think the main reason was due to current differences in the typing of the interface vs the class methods, it can cause issues with code analysis. But what you're saying makes sense actually so I think I'll revert this and those differences will just need to be fixed

Reply:
Fair enough.